src/range.rs

```diff
@@ -0,0 +1,50 @@
+//! This module abstracts over ranges to allow functions in the crate to take
+//! ranges as input in an ergonomic way. It howvers seals this abstractions to
+//! leave full control in this crate.
+
+/// Abstract over `std::ops::{Range,RangeFrom,RangeTo,RangeFull}`
```

Comment:
Any reason to leave out `RangeInclusive` and `RangeToInclusive`?

Reply:
No, I just forgot about them!

Reply:
I'm wondering now, should `Range::start` trait return an `usize` instead of an `Option<usize>`, with `RangeTo`, `RangeFull` returning `0` in that case? I'm wondering if there can be a downside. Right now I don't see a case where I'd like to know that the start of the range was not explicitly specified.

Reply:
You have don't have negative or positive starting offsets so returning an `usize` should be OK

Reply:
`RangeInclusive` and `RangeToInclusive` are now implemented, and `Range::start` returns an usize.